errors.go

```diff
@@ -59,6 +59,11 @@ type Error struct {
 	// exported for serialization, but you should use Retryable to read the value.
 	IsRetryable *bool `json:"is_retryable"`
 
+	// Incremented each time the error is marshalled so that we can tell (approximately) how many services the error
+	// has propagated through.  Higher level code can use this to influence decisions, for example it may only be
+	// desirable to retry on an error that's only been marshalled once to avoid retries on top of retries... ad nauseam
+	MarshallCount int `json:"marshall_count"`
```

Comment:
Without thinking too hard about the overall approach, an immediate nit is: shouldn't it be 'marshal' for a present tense usage like this? I was under the impression the double-l is only used in the past tense 🤔

Reply:
Yes, you're right.  I'll update 👍